app-misc/pip3line/pip3line-3.4.2.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit cmake-utils python-r1
+
+DESCRIPTION="Raw bytes manipulation, transformations (decoding and more) and interception"
+HOMEPAGE="https://github.com/metrodango/pip3line"
+
+KEYWORDS=""
+SRC_URI=""
+
+if [[ ${PV} == 9999* ]] ; then
+    inherit git-r3
+    EGIT_REPO_URI="https://github.com/metrodango/pip3line.git"
+    EGIT_BRANCH="master"
+else
+    SRC_URI="https://github.com/metrodango/pip3line/archive/v${PV}.tar.gz  -> ${P}.tar.gz"
+    KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="distorm openssl qscintilla"
+
+RDEPEND="
+    ${PYTHON_DEPS}
+    dev-qt/qtconcurrent:5
+    dev-qt/qtcore:5
+    dev-qt/qtgui:5
+    dev-qt/qtnetwork:5
+    dev-qt/qtsvg:5
+    dev-qt/qtwidgets:5
+    dev-qt/qtxmlpatterns:5
+    distorm? ( dev-libs/distorm64 )
+    openssl? ( dev-libs/openssl:0= )
+    qscintilla? ( x11-libs/qscintilla )"
+
+DEPEND="${RDEPEND}"
+
+src_configure() {
+
+    if [[ "${PYTHON_TARGETS}" == *python3* ]]; then
+        python3_usage=yes
+    else
+        python3_usage=no
+    fi
+
+    local mycmakeargs=(
+        -DBASIC=yes
+        -DWITH_OPENSSL=$(usex openssl)
+        -DWITH_PYTHON27=$(usex python_targets_python2_7)
+        -DWITH_PYTHON3=${python3_usage}
+        -DWITH_SCINTILLA=$(usex qscintilla)
+        -DWITH_DISTORM=$(usex distorm)
+    )
+
+    cd ext || die
+    ./gen_distorm.sh || die
+
+    cmake-utils_src_configure
+}
+
+pkg_postinst() {
+    if use distorm ; then
+        echo
+        ewarn "The Distorm plugin was compiled."
+        ewarn "Beware the current distorm64 installation does not"
+        ewarn "copy the necessary files at the right place."
+        ewarn "The native library is in :"
+        ewarn "/usr/lib64/python[version]/site-packages/distorm3/libdistorm3.so"
```

Comment:
This seems fishy to me. If you build against the system distorm, users shouldn't need to do this kind of stuff.

Reply:
One can't build against the system distorm64 because none of the headers are installed by the current package.

 * Searching for distorm64 in dev-libs ...
 * Contents of dev-libs/distorm64-3.3.4:
/usr
/usr/lib64
/usr/lib64/python2.7
/usr/lib64/python2.7/site-packages
/usr/lib64/python2.7/site-packages/distorm3
/usr/lib64/python2.7/site-packages/distorm3-3.3.4-py2.7.egg-info
/usr/lib64/python2.7/site-packages/distorm3/__init__.py
/usr/lib64/python2.7/site-packages/distorm3/__init__.pyc
/usr/lib64/python2.7/site-packages/distorm3/__init__.pyo
/usr/lib64/python2.7/site-packages/distorm3/libdistorm3.so
/usr/lib64/python2.7/site-packages/distorm3/sample.py
/usr/lib64/python2.7/site-packages/distorm3/sample.pyc
/usr/lib64/python2.7/site-packages/distorm3/sample.pyo
/usr/lib64/python3.6
/usr/lib64/python3.6/site-packages
/usr/lib64/python3.6/site-packages/distorm3
/usr/lib64/python3.6/site-packages/distorm3-3.3.4-py3.6.egg-info
/usr/lib64/python3.6/site-packages/distorm3/__init__.py
/usr/lib64/python3.6/site-packages/distorm3/__pycache__
/usr/lib64/python3.6/site-packages/distorm3/__pycache__/__init__.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/distorm3/__pycache__/__init__.cpython-36.pyc
/usr/lib64/python3.6/site-packages/distorm3/__pycache__/sample.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/distorm3/__pycache__/sample.cpython-36.pyc
/usr/lib64/python3.6/site-packages/distorm3/libdistorm3.so
/usr/lib64/python3.6/site-packages/distorm3/sample.py
/usr/share
/usr/share/doc
/usr/share/doc/distorm64-3.3.4
/usr/share/doc/distorm64-3.3.4/README.md.bz2